crates/metaslang/cst/src/text_index.rs

```diff
@@ -1,22 +1,46 @@
 use std::fmt::Display;
-use std::ops::{Add, AddAssign, Range, Sub, SubAssign};
+use std::ops::{Add, AddAssign, Range};
 
 use serde::Serialize;
 
-#[derive(Default, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
+#[derive(Default, Hash, Copy, Clone, PartialEq, Eq, Debug, Serialize)]
 pub struct TextIndex {
```

Comment:
I'm guessing we're bundling everything into a single type for convenience, lack of perf numbers and velocity but I imagine that having just the existing TextIndex, without having to always track/calculate the column, that allows to quickly slice into the contents buffer will be useful (we might emulate this with `line: 0` but that's not technically correct).

Do you think we should have a separate `TextPosition` that might bundle the the existing/hypothetical `TextIndex` with the line?

Reply:
This is a decision dependent on perf analysis IMO.